src/server/api/AdminPanel.ts

```diff
@@ -0,0 +1,187 @@
+		res.status(403).send("Cannot send commands without the admin role");
+		return;
+	}
+	// TODO prevent affecting accounts with equal or higher roles
+	switch (commandAndArgs[0]) {
+		case "ban":
+			return;
+		case "delete":
+			deleteCommand(command, commandAndArgs, res);
+			return;
+		case "username":
+			usernameCommand(command, commandAndArgs, res);
+			return;
+		case "logout":
+			logoutUser(command, commandAndArgs, res);
+			return;
+		case "verify":
+			return;
+		case "post":
+			return;
+		case "invites":
+			return;
+		case "announce":
+			return;
+		case "userinfo":
+			getUserInfo(command, commandAndArgs, res);
+			return;
+		case "help":
+			helpCommand(commandAndArgs, res);
+			return;
+		default:
+			res.status(422).send("Unknown command.");
+			return;
+	}
+}
+
+function deleteCommand(command: string, commandAndArgs: string[], res: Response) {
+	logCommand(command);
+	if (commandAndArgs.length < 2) {
+		res.status(422).send("Invalid number of arguments, expected 1, got " + (commandAndArgs.length - 1) + ".");
+		return;
+	}
+	try {
+		deleteAccount(getMemberDataByCriteria(["user_id"], "username", commandAndArgs[1])["user_id"] ?? (() => { throw new Error(); })(), commandAndArgs[2] ?? "");
+	}
+	catch {
+		res.status(422).send("User " + commandAndArgs[1] + " does not exist.");
+		return;
+	}
+	res.status(200).send("");
+}
+
+function usernameCommand(command: string, commandAndArgs: string[], res: Response) {
+	if (commandAndArgs[1] === "get") {
+		if (commandAndArgs.length < 3) {
+			logCommand(command);
+			res.status(422).send("Invalid number of arguments, expected 2, got " + (commandAndArgs.length - 1) + ".");
+			return;
+		}
+		const parsedId = Number.parseInt(commandAndArgs[2]!);
+		logEvents("Command executed: " + command + "\nResult: " + getMemberDataByCriteria(["username"], "user_id", parsedId)["username"] + "\n", "adminCommands");
+		const username = getMemberDataByCriteria(["username"], "user_id", parsedId)["username"];
+		res.status(username === undefined ? 422 : 200).send(username ?? "User with id " + parsedId + " does not exist.");
+	}
+	else if (commandAndArgs[1] === "set") {
+		if (commandAndArgs.length < 4) {
+			res.status(422).send("Invalid number of arguments, expected 3, got " + (commandAndArgs.length - 1) + ".");
+			return;
+		}
+		// TODO add username changing logic
+		res.status(503).send("Changing usernames is not yet supported.");
+	}
+	else {
+		res.status(422).send("Invalid subcommand, expected either get or set, got " + commandAndArgs[1] + ".");
+	}
+}
+
+function logoutUser(command: string, commandAndArgs: string[], res: Response) {
+	logCommand(command);
+	if (commandAndArgs.length < 2) {
+		res.status(422).send("Invalid number of arguments, expected 1, got " + (commandAndArgs.length - 1) + ".");
+		return;
+	}
+	try {
+		deleteAllSessionsOfUser(getMemberDataByCriteria(["user_id"], "username", commandAndArgs[1])["user_id"] ?? (() => { throw new Error(); })());
```

Comment:
Place `commandAndArgs[1]` into a `username` variable.
Avoid the try catch block by setting the results of getMemberDataByCriteria() to a variable. Then if the user_id is undefined we can return an error that the user doesn't exist.